front/app/components/ProjectAndFolderCards/index.tsx

```diff
@@ -21,23 +23,35 @@ export type TLayout = 'dynamic' | 'threecolumns';
 export interface Props {
   showTitle: boolean;
   layout: TLayout;
-  publicationStatusFilter: PublicationStatus[];
   showSearch?: boolean;
   currentlyWorkingOnText?: Multiloc;
 }
 
+interface InnerProps extends Props {
+  statusCountsWithoutFilters: IStatusCounts;
+}
+
+export const PUBLICATION_STATUSES: PublicationStatus[] = [
+  'published',
+  'archived',
+];
+
 const ProjectAndFolderCards = ({
-  publicationStatusFilter,
   showSearch = false,
+  statusCountsWithoutFilters,
   ...otherProps
-}: Props) => {
+}: InnerProps) => {
+  const allStatusCountsWithoutFilters = getStatusCounts(
+    statusCountsWithoutFilters
+  );
+
   // used locally to keep track of the depth of the search
   const [search, setSearch] = useState<string | null>(null);
   const [topicIds, setTopicsIds] = useState<string[] | null>(null);
   const [areaIds, setAreasIds] = useState<string[] | null>(null);
-  const [publicationStatus, setPublicationStatus] = useState<
-    PublicationStatus[]
-  >(publicationStatusFilter);
+  const [currentTab, setCurrentTab] = useState<PublicationTab>(
+    getCurrentTab(allStatusCountsWithoutFilters)
```

Comment:
The main change is that we first fetch the 'status counts' (i.e. how many admin publications are published, archived etc) and based on that decide the correct tab (e.g. published, archived, all) BEFORE requesting the rest of the admin publications.

Before, we were requesting the admin publications, and then checking in the `useEffect` nightmare spaghetti (see above) whether we needed to re-request admin publications in case we were on the wrong tab

Reply:
This is so much clearer.